pkg/runner/runner.go

```diff
@@ -278,3 +298,67 @@ func recordStateMessage(state *engine.State) error {
 	filename := filepath.Join(tmpdir, fmt.Sprintf("gptscript-state-%v-%v", hostname, time.Now().UnixMilli()))
 	return os.WriteFile(filename, data, 0444)
 }
+
+func (r *Runner) handleCredentials(callCtx engine.Context, monitor Monitor, env []string) ([]string, error) {
+	c, err := config.ReadCLIConfig("")
+	if err != nil {
+		return nil, fmt.Errorf("failed to read CLI config: %w", err)
+	}
+
+	store, err := credentials.NewStore(c, r.credCtx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create credentials store: %w", err)
+	}
+
+	for _, credToolName := range callCtx.Tool.Credentials {
+		cred, exists, err := store.Get(credToolName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get credentials for tool %s: %w", credToolName, err)
+		}
+
+		// If the credential doesn't already exist in the store, run the credential tool in order to get the value,
```

Comment:
Not sure if that'd be too much, but maybe we can have a flag for "reprompt" to update all cached credentials in a run? My laziness coming through.. :D

Reply:
I asked Darren about it and he wants to hold off on this for now.